optimade/validator/validator.py

```diff
@@ -712,7 +732,18 @@ def _construct_single_property_filters(
                 )
 
             response = response.json()
-            num_data_returned[operator] = response["meta"]["data_returned"]
+
+            if "meta" not in response or "more_data_available" not in response["meta"]:
+                raise ResponseError(
+                    f"Required field `meta->more_data_available` missing from response for {request}."
+                )
+
+            if not response["meta"]["more_data_available"]:
+                num_data_returned[operator] = len(response["data"])
+            else:
+                num_data_returned[operator] = response["meta"].get("data_returned")
```

Comment:
Again, can't remember if this by default returns `None` or if it raises `KeyError`.
```suggestion
                num_data_returned[operator] = response["meta"].get("data_returned", None)
```